src/cosalib/metal.py

```diff
@@ -0,0 +1,319 @@
+        Returns the rootfs size and the image size.
+        """
+        size = self.ostree_repo_estimated_size
+        return f"{int(size * 1.35)}M"
+
+    @property
+    def disk_size(self):
+        """
+        Returns the size of the disk. This is the size of the
+        rootfs with padding
+        """
+        size = self.img_cfg.get("size")
+        if size:
+            if str(size).upper().endswith("M"):
+                return size
+            else:
+                return f"{size}G"
+        return f"{self.rootfs_size + 513}M"
+
+    @property
+    def rootfs_type(self):
+        rootfs = self.img_cfg.get("rootfs", DEFAULT_ROOTFS)
+        rootfs_luks = self.img_cfg.get("luks_rootfs", "None").lower()
+        if rootfs_luks in ("yes", "1", "true"):
+            rootfs = "luks"
+            log.debug("rootfs will be housed in LUKS")
+        return rootfs
+
+    def get_disk_args(self, ref):
+        disk_args = []
+
+        tmod_cmd = run_verbose(
+            [
+                "ostree", f"--repo={self.ostree_repo}",
+                "ls", ref, "/usr/lib/modules",
+            ],
+            check=True, capture_output=True
+        ).stdout.decode('utf-8').splitlines()[-1]
+        target_moduledir = tmod_cmd.split()[-1]
+        log.info(f"Kernel Modules found at {target_moduledir}")
+
+        kconfig = run_verbose(
+            [
+                "ostree", f"--repo={self.ostree_repo}",
+                "cat", ref,
+                f"{target_moduledir}/config"
+            ],
+            check=True, capture_output=True
+        ).stdout.decode('utf-8')
+
+        disk_args = []
+        for l in kconfig.splitlines():
+            if "CONFIG_FS_VERITY=y" in l:
+                disk_args = disk_args.append('--boot-verity')
+                break
+        if '--boot-verity' not in disk_args:
+            log.info(('NOTE: Missing CONFIG_FS_VERITY from target kernel'
+                      ' config'))
+
+        return disk_args
+
+    def mk_image(self):
+        """
+        mk_image creates the final disk
+        """
+        self.create_ref()
+        # get the root files system
+
+        # Get the kernel args
+        kargs = self.kargs
+        kargs.extend(self.img_cfg.get("extra-kargs", []))
+        kargs = [x.format(
+                    platform=self.platform,
+                    terminal=arch_terminal(self.basearch)
+        ) for x in kargs]
+        log.debug(f"using '{kargs}' for image kernel cli")
+
+        # Create the ref
+        self.create_ref()
+        ref = self.get_rev(self.tmpref_name)
+
+        # create the disk
+        run_verbose([
+            'qemu-img', 'create', '-f',
+            self.format, self.image_path,
+            self.disk_size
+        ])
+
+        log.info(f"Found ostree ref: '{ref}'")
+        log.info((f"Creating {self.disk_size} disk with {self.rootfs_size}"
+                  f" rootfs to {self.image_name}"))
+
+        # Format the command
+        cmd = ['runvm']
```

Comment:
This is a toplevel script/entrypoint now?  I don't see it in master or this PR?

Reply:
It is not -- its "sourced". `cmldib.py` was changed to allow using the `cmdlib.sh`. See:https://github.com/darkmuggle/coreos-assembler/commit/840080a255e7ed4036c5a5a6ebc38fae520e74ee#diff-f544d2b85a345113cb6b0bd75281fe86R65